pkgs/development/compilers/llvm/3.8/llvm.nix

```diff
@@ -81,6 +85,14 @@ in stdenv.mkDerivation rec {
     ++ stdenv.lib.optionals ( isDarwin) [
     "-DLLVM_ENABLE_LIBCXX=ON"
     "-DCAN_TARGET_i386=false"
+  ] ++ stdenv.lib.optionals stdenv.hostPlatform.isMusl [
+    "-DLLVM_HOST_TRIPLE=${stdenv.hostPlatform.config}"
```

Comment:
Can we handle these without the condition? I guess that could break things but I would much prefer to have these 3 flags be unconditional.

Reply:
I would too -- but previously was asked to keep them musl-only for now so I was following along here.  How about I make a new PR doing this across LLVM versions containing this?

Reply:
Okay that's definitely fine to make it musl-only. I just want to minimize conditionals in Nixpkgs as much as possible.

Reply:
I assume since the `nativeBuildInput` stuff is done in here, doing that too wouldn't make it a mass rebuild, right? I'd ping whoever asked before, either in this PR, or a 3rd master PR (not the 4 5 6 one).

Reply:
I'll make a new PR if that's alright with you-- rather keep unrelated things separate even if it looks like we have 3 PR's on that subject now but oh well :).

Reply:
(since that PR will conflict with this one if not based on top, I'll wait to put that together until this is merged since I think there's little controversial/risky here, esp in comparison :))